app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileScreen.kt

```diff
@@ -0,0 +1,406 @@
+            modifier = Modifier.width(100.dp).testTag("cancelButton")) {
+              Text(text = "Cancel", color = md_theme_light_error)
+            }
+      }
+}
+
+/**
+ * Display the profile of the user, with the possibility to edit it
+ *
+ * @param profile the profile of the user
+ * @param onProfileChange enable to transmit the changed to the caller
+ * @param viewModel the viewModel that will handle the profile
+ * @author Menzo Bouaissi
+ * @since 1.0
+ * @last update 1.0
+ */
+@Composable
+fun EditableVisitCard(
+    profile: Profile,
+    onProfileChange: (Profile) -> Unit,
+    viewModel: ProfileViewModel
+) {
+  val profile2 = profile.copy()
+  var firstName by remember { mutableStateOf(profile2.firstName) }
+  var lastName by remember { mutableStateOf(profile2.lastName) }
+  var description by remember { mutableStateOf(profile2.description) }
+
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    ImageSelection(profile = profile2, onImageChange = { uri -> profile2.profilePictureUri = uri })
+    EditableTextFields(
+        firstName = firstName,
+        lastName = lastName,
+        description = description,
+        onFirstNameChange = { newName ->
+          if (newName.length <= MAX_NBR_CHAR_NAMES) firstName = newName
+        },
+        onLastNameChange = { newName ->
+          if (newName.length <= MAX_NBR_CHAR_NAMES) lastName = newName
+        },
+        onDescriptionChange = { newDescription ->
+          if (newDescription.length <= MAX_NBR_CHAR_DESC) description = newDescription
+        })
+    Spacer(Modifier.padding(18.dp))
+    ActionButtons(
+        onSave = {
+          viewModel.toggleEditMode()
+          onProfileChange(
+              profile.copy(
+                  firstName = firstName,
+                  lastName = lastName,
+                  description = description,
+                  profilePictureUri = profile2.profilePictureUri))
+        },
+        onCancel = { viewModel.toggleEditMode() })
+  }
+}
+
+/**
+ * Handle the logic behind the image, when the user has a profile picture, we display it via
+ * AsynImage, else we just display a based image with Image
+ *
+ * @param modifier the modifier to be applied to the Icon
+ * @param profile the profile of the user
+ * @author Menzo Bouaissi
+ * @since 1.0
+ * @last update 1.0
+ */
+@Composable
+fun SelectImage(modifier: Modifier, profile: Profile) {
+  if (profile.profilePictureUri != null) {
+    AsyncImage(
+        model = profile.profilePictureUri,
+        contentDescription = "Profile picture",
+        modifier = modifier)
+  } else {
+    Image(
+        painter = painterResource(id = R.drawable.profile_picture),
+        contentDescription = "Profile picture",
+        modifier = modifier)
+  }
+}
+
+/**
+ * For debug use, to see the screen
+ *
+ * @author Menzo Bouaissi
+ * @since 1.0
+ * @last update 1.0
+ */
+@Preview
+@Composable
+fun ProfileScreenPreview() {
+  ProfileScreen(ProfileViewModel())
+}
```

Comment:
I think from the maintainability perspective it's better to extract the components out. I really like the documentation and we should adopt the habit as a team.

Reply:
No I meant for the overall function I think it is better to define new files inside ui.components.profile and make the Profile screen shorter. Sorry for the confusion.